src/client/java/minicraft/item/SummonItem.java

```diff
@@ -68,12 +68,15 @@ public boolean interactOn(Tile tile, Level level, int xt, int yt, Player player,
 					// If the player nears the center.
 					if (new Rectangle(level.w/2-3, level.h/2-3, 7, 7).contains(player.x >> 4, player.y >> 4)) {
 						if (!ObsidianKnight.active) {
-
-							// Pay stamina
-							if (player.payStamina(2)) {
-								level.add(new KnightStatue(5000), level.w/2, level.h/2, true);
-								Logger.tag("SummonItem").debug("Summoned new Knight Statue");
-								success = true;
+							if (level.getEntitiesOfClass(KnightStatue.class).length == 0) { // Prevent unintended behaviors
```

Comment:
This is a strange way of checking for other statues. Are you sure there isn't a better way to do this?

Reply:
I have changed to use for loop.

Reply:
Yeah, then it only checks until it finds one, and not the whole array 100% of the time.